src/registry/mod.rs

```diff
@@ -14,44 +14,161 @@
 
 use std::{
     fmt::{self, Display},
-    ops::{Deref, DerefMut},
     str::FromStr,
 };
 
 mod artifactory;
 #[cfg(test)]
 mod cache;
 
-pub use artifactory::Artifactory;
+use crate::manifest::DependencyManifest;
+use crate::{config, manifest::Dependency};
+pub use artifactory::{Artifactory, CertValidationPolicy};
 use miette::{ensure, miette, Context, IntoDiagnostic};
 use semver::VersionReq;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 use url::Url;
 
-use crate::manifest::{Dependency, DependencyManifest};
-
 /// A representation of a registry URI
 #[derive(Debug, Clone, Hash, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
 pub struct RegistryUri(Url);
 
+impl RegistryUri {
+    /// Get the host component of the registry URI
+    pub fn host(&self) -> Option<&str> {
+        self.0.host_str()
+    }
+
+    /// Get the path component of the registry URI
+    pub fn path(&self) -> &str {
+        self.0.path()
+    }
+}
+
+/// A reference to a registry
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum RegistryRef {
+    /// A URL to a registry
+    Url(RegistryUri),
+    /// An alias to a registry
+    Alias(String),
+    /// A resolved alias to a registry
+    ResolvedAlias {
```

Comment:
Thus this case seems both type unsafe and unnecessary to me as you would want to go from RegistryRef to RegistryUri during alias resolution